lang/c++/include/avro/buffer/BufferStreambuf.hh

```diff
@@ -135,7 +135,11 @@ protected:
                 memcpy(c, gptr(), toCopy);
                 c += toCopy;
                 bytesCopied += toCopy;
-                gbump(toCopy);
+                while (toCopy > INT_MAX) {
```

Comment:
I'd prefer using [numeric_limits](https://en.cppreference.com/w/cpp/types/numeric_limits/max) here as using macros can be fragile.

Reply:
I did that first, switched to the macro when gcc 9.4 on the ARM agent was complaining about signedness... I can add a `static_cast` instead, hopefully newer compilers don't flag that as a useless cast.

Reply:
Ah I see... thanks!
You can leave it as it is now and add comment that this is only needed for old gcc compatibility.

We do need to upgrade gcc on ARM runner at some point. FYI @martin-g, that's not the first time this one bites

Reply:
ASF Infra team promised to upgrade the Github builders to a newer Ubuntu, but there is no timeframe ...